packages/ui-components/src/components/CodeBlock/CodeBlock.test.tsx

```diff
@@ -8,28 +8,28 @@ import { render, screen } from "@testing-library/react"
 import { CodeBlock } from "./index"
 
 describe("CodeBlock", () => {
```

Comment:
Would be great during these refactors to improve testing. Some missing edge cases could be:
- Component rendering when both `content` and `children` are `null` or `undefined`.
- Rendering (fallback and default) behaviour for different `lang` prop values, including unsupported languages.
- Clipboard copy functionality to ensure correct text is copied and the "Copied!" message appears.
- Ensuring `timeoutRef` is properly cleared when the component unmounts to prevent memory leaks.
- Ensuring `theCode.current` is correctly assigned and not accessed when `null`.